intranet/apps/dashboard/views.py

```diff
@@ -255,39 +256,43 @@ def get_announcements_list(request, context):
             midnight = timezone.localtime().replace(hour=0, minute=0, second=0, microsecond=0)
             events = Event.objects.visible_to_user(user).filter(time__gte=midnight, show_on_dashboard=True)
 
-    items = sorted(chain(announcements, events), key=lambda item: (item.pinned, item.added))
-    items.reverse()
+    def announcements_sorting_key(item: Announcement):
+        nonlocal context
+        if context["show_expired"]:
+            # sort items first by expiration date then by added date
+            return (item.expiration_date, item.added)
+        # otherwise sort by pinned and then added date
+        return (item.pinned, item.added)
+
+    items = sorted(chain(announcements, events), key=announcements_sorting_key, reverse=True)
 
     return items
 
 
 def paginate_announcements_list(request, context, items):
     """
-    ***TODO*** Migrate to django Paginator (see lostitems)
+    Paginate ``items`` in groups of 15
 
     """
 
-    # pagination
-    if "start" in request.GET:
-        try:
-            start_num = int(request.GET.get("start"))
-        except ValueError:
-            start_num = 0
-    else:
-        start_num = 0
-
-    display_num = 15
-    end_num = start_num + display_num
-    prev_page = start_num - display_num
-    more_items = (len(items) - start_num) > display_num
     try:
-        items_sorted = items[start_num:end_num]
-    except (ValueError, AssertionError):
-        items_sorted = items[:display_num]
-    else:
-        items = items_sorted
+        page_num = int(request.GET["page"])
+    except (ValueError, KeyError):
+        page_num = 1
```

Comment:
Using exceptions instead of actual conditional logic here seems weird. Instead of a KeyError when accessing page, you can just use the .get() method on a dictionary with a default value. Instead of the ValueError, you can check if the string is a number using .isdigit()